langchain/callbacks/tracers/base.py

```diff
@@ -189,6 +189,38 @@ def on_llm_error(
         llm_run.end_time = datetime.utcnow()
         self._end_trace(llm_run)
 
+    def on_chat_model_start(
+        self,
+        serialized: Dict[str, Any],
+        messages: List[List[BaseMessage]],
+        *,
+        run_id: UUID,
+        parent_run_id: Optional[UUID] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Start a trace for an LLM run."""
+        if self.session is None:
+            self.session = self.load_default_session()
+
+        run_id_ = str(run_id)
+        parent_run_id_ = str(parent_run_id) if parent_run_id else None
+
+        execution_order = self._get_execution_order(parent_run_id_)
+        # Duplicate information until we break compatibility with the old API
+        prompts = [get_buffer_string(batch) for batch in messages]
```

Comment:
@agola11 I know you wanted to avoid serializing then deserializing. I'm less cognizant of consumers of this method right now but it seems like changing the type of LLMRun would break some implementations, so may as well go the whole way if we do that (e.g. https://github.com/wandb/wandb/blob/6fe4456cee5005408a3e74d3999e2cbfb761081b/wandb/integration/langchain/util.py#L129)

Reply:
I wouldn't worry too much about that. They've specified that their package only works with older versions of langchain. Let's not serialize and deserialize, it's messy and overcomplicates things. I would just use `get_buffer_string` to serialize as string in the langchainv1 tracer and send as messages in the newer langchain tracer. I'll communicate with wandb